src/global/project/mod.rs

```diff
@@ -605,18 +607,51 @@ impl Project {
         Ok(state_changes)
     }
 
+    /// Get all installed executables for specific environment.
+    pub async fn executables(
+        &self,
+        env_name: &EnvironmentName,
+    ) -> miette::Result<IndexMap<PackageName, Vec<(String, PathBuf)>>> {
+        let parsed_env = self
+            .environment(env_name)
+            .ok_or_else(|| miette::miette!("Environment {} not found", env_name.fancy_display()))?;
+
+        let package_names: Vec<_> = parsed_env.dependencies().keys().cloned().collect();
+
+        let mut executables_for_package = IndexMap::new();
+
+        for package_name in &package_names {
+            let prefix = self.environment_prefix(env_name).await?;
+            let prefix_package = prefix.find_designated_package(package_name).await?;
+            let mut package_executables = prefix.find_executables(&[prefix_package]);
+
+            // Sometimes the package don't ship executables on their own.
+            // We need to search for it in different packages.
+            if !package_executables
+                .iter()
+                .any(|(exec_name, _)| exec_name.as_str() == package_name.as_normalized())
+            {
+                if let Some(exec) = find_binary_by_name(&prefix, package_name).await? {
+                    package_executables.push(exec);
+                }
+            }
+
+            executables_for_package.insert(package_name.clone(), package_executables);
+        }
+        Ok(executables_for_package)
+    }
+
     /// Find the exposed names that are no longer installed in the environment
     /// and remove them.
     /// This needs to happen after a different version of a package was installed
     /// which doesn't have the same executables anymore.
-    pub async fn remove_broken_expose_names(
+    pub async fn sync_exposed_names(
```

Comment:
If the behavior of this function changed, maybe the doc string needs to be adapted as well?